lib/fog/ovirt/requests/compute/v4/create_vm.rb

```diff
@@ -27,9 +27,15 @@ def process_vm_opts(opts)
           # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
           def create_vm(attrs)
             attrs = attrs.dup
-            attrs[:cluster_name] ||= datacenter.clusters.first.name unless attrs[:cluster]
+
+            unless attrs[:cluster]
+              cluster = client.system_service.clusters_service.list(:search => "datacenter=#{datacenter_hash[:name]}").first
+              attrs[:cluster] = cluster
+              attrs[:cluster_name] ||= cluster.name
```

Comment:
Can there be a situation where `attrs[:cluster]` is unset, but `attrs[:cluster_name]` is set?